src/libraries/System.Text.Json/tests/Serialization/WriteValueTests.cs

```diff
@@ -68,5 +68,38 @@ public static void SerializeToWriterRoundTripEscaping()
 
             Assert.Equal(normalizedString, json);
         }
+
+        public class CustomClassToExceedMaxBufferSize
+        {
+            private static readonly string s_name = new string('a', 100_000_000);//Large enough value to cause integer overflow exception when allocating buffer and small enought to not cause a "The JSON value of length X is too large and not supported."
+            public string GetName1 => s_name;
+            public string GetName2 => s_name;
+            public string GetName3 => s_name;
+            public string GetName4 => s_name;
+            public string GetName5 => s_name;
+            public string GetName6 => s_name;
+            public string GetName7 => s_name;
+            public string GetName8 => s_name;
+            public string GetName9 => s_name;
+            public string GetName10 => s_name;
+            public string GetName11 => s_name;
+            public string GetName12 => s_name;
+            public string GetName13 => s_name;
+            public string GetName14 => s_name;
+            public string GetName15 => s_name;
+            public string GetName16 => s_name;
+            public string GetName17 => s_name;
+            public string GetName18 => s_name;
+            public string GetName19 => s_name;
+            public string GetName20 => s_name;
```

Comment:
You don't need to use up 4 GB to test the scenario. Anything over ~2GB would be sufficient.

Keeping 11 copies of the string would work as well (you can remove the `GetName12` to `GetName20`).

Reply:
If I reduce it, this code is not executed:

![image](https://user-images.githubusercontent.com/1336227/74621179-ca9ecf80-50f0-11ea-83d8-015b8fa4e3b1.png)

Reply:
> The path that is likely not covered (and this sequence of code flow won't show up in test coverage):
> `(uint)(currentLength + growBy) > int.MaxValue` is **true** but `(uint)(currentLength + sizeHint) > int.MaxValue` is **false**

We can cover that path, using the class with less than 20 name fields, but I'm not sure if we need it.

Reply:
> but I'm not sure if we need it.

It wouldn't hurt to add a test for that as well. It does improve our "path" coverage, in case the grow logic were to change.

Reply:
Actually, with 20 Fields, both branches are covered:

![image](https://user-images.githubusercontent.com/1336227/74903384-9bd96100-535d-11ea-8c12-600d07d9bd05.png)

Reply:
Nice.